karapace/karapace_all.py

```diff
@@ -11,29 +10,17 @@
 import sys
 
 
-class KarapaceAll(KafkaRest, KarapaceSchemaRegistry, KarapaceBase):
+class KarapaceAll(KafkaRest, KarapaceSchemaRegistry):
     # pylint: disable=super-init-not-called
-    def __init__(self, config_file_path: str, config: dict) -> None:
-        KarapaceBase.__init__(self, config_file_path=config_file_path, config=config)
-        KafkaRest._init(self, config=config)
-        KafkaRest._add_routes(self)
-        KarapaceSchemaRegistry._init(self, config=config)
-        KarapaceSchemaRegistry._add_routes(self)
+    def __init__(self, config: dict) -> None:
+        super().__init__(config=config)
         self.log = logging.getLogger("KarapaceAll")
         self.app.on_shutdown.append(self.close_by_app)
 
     def close_by_app(self, app):
         # pylint: disable=unused-argument
         self.close()
 
-    def close(self):
-        for cls in [KarapaceBase, KafkaRest, KarapaceSchemaRegistry]:
-            try:
-                cls.close(self)
-            except:  # pylint: disable=bare-except
-                pass
-            KafkaRest.close_producers(self)
```

Comment:
After these changes `close_producers` is no longer called at all as far as I can see (except in integration tests)? Should it be called or eliminated altogether if no longer needed?

Of course existing code is a bit oddity as it is calling async method here without waiting for completion.

Reply:
Good catch. Since that is part of `KafkaRest` IMO it done on `KafkaRest.close`. And we definitely should wait on it. This code here was just creating a coroutine that was never executed, so it had no effect.

It also looks like we are not calling `close` in the main method. It seems this method was never called.

Reply:
Isn't `close`method called from `close_by_app` which is triggered by signal handler?

Reply:
> Isn't closemethod called from close_by_app which is triggered by signal handler?

Yep, you're right. [Here is the docs](https://docs.aiohttp.org/en/stable/web_advanced.html?highlight=on_shutdown#graceful-shutdown)

Reply:
So, the code example in the docs uses an async function. So I converted the `close` methods to async and now we can call `close_producers` there.